### PR TITLE
Add 35% coverage tests of the feature

### PR DESCRIPTION
**Pull request by zancas**
@mattgreene 
#### What issues does this address?

Issues:
#109 Disconnected Service Deployment

Problem: Disconnected service lacked tests.

Analysis: This commit adds 4 passing tests.  The 4 skipped tests are skipped
because of known bugs.  The passing tests cover 35% of the agent code.

* test_featureoff_nosegid_lb: This test will not pass until the feature short
circuits prior to l2 configuration.  See issue #242

* test_nosegid_lb_timeout:  This test fails with an invalid vxlan ID error.
The service object used in the test should not cause the agent to exercise
l2 network creation code. See issue #242

* test_segid_listener: The agent appears to cache state across
icontroldevice instantiations, and therefore across tests.  I'm going to
add tests to liberty to further describe this bug. This is documented in
issue #243

* test_route_domain_naming: wip

Tests: test_disconnected_service_creation.py

### CODE REVIEW

File: test/functional/neutronless/l2adjacent/Dockerfile [hunk 1]
```diff
@@ -22,4 +22,4 @@ RUN pip install --upgrade /root/devenv/f5-openstack-agent
 COPY ./f5-common-python/ /root/devenv/f5-common-python/
 RUN pip install --upgrade /root/devenv/f5-common-python/
 WORKDIR /root/devenv/f5-openstack-agent/test/functional/neutronless/l2adjacent
-RUN py.test -s -v --symbols /root/devenv/f5-openstack-agent/test/functional/neutronless/l2adjacent/conf_symbols.json -v /root/devenv/f5-openstack-agent/test/functional/neutronless/l2adjacent/test_loadbalancer_create.py
+RUN py.test --cov /root/devenv/f5-openstack-agent/f5_openstack_agent -v --symbols /root/devenv/f5-openstack-agent/test/functional/neutronless/l2adjacent/conf_symbols.json -v /root/devenv/f5-openstack-agent/test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/Dockerfile]: Make RUN work on the entire directory and add a comment line with how to run a single suite.

File: test/functional/neutronless/l2adjacent/conf_symbols.json [hunk 2]
```diff
@@ -3,10 +3,10 @@
   "tenant_password":                              "changeme",
   "tenant_name":                                  "testlab",
   "auth_url":                                     "http://OSCONTROLLER_IP:5000/v2.0",
-  "bigip_ip":                                     "DEVICE_IP",
+  "bigip_ip":                                     "10.190.7.130",
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/conf_symbols.json]: Revert this change.

File: test/functional/neutronless/l2adjacent/conf_symbols.json [hunk 3]
```diff
@@ -3,10 +3,10 @@
...
   "bigip_username":                               "admin",
   "bigip_password":                               "admin",
-  "icontrol_hostname":                            "DEVICE_IP",
+  "icontrol_hostname":                            "10.190.7.130",
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/conf_symbols.json]: Revert this change.

File: test/functional/neutronless/l2adjacent/oslo_confs.json [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+        "f5_snat_addresses_per_subnet": 1, 
+        "f5_snat_mode": true, 
+        "f5_vtep_folder": "Common", 
+        "f5_vtep_selfip_name": "selfip.client", 
+        "force_gateway_on_subnet": true, 
+        "host": "host-3.int.lineratesystems.com", 
+        "icontrol_config_mode": "objects", 
+        "icontrol_connection_retry_interval": 10, 
+        "icontrol_connection_timeout": 30, 
+        "icontrol_hostname": "10.190.7.130",
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/oslo_confs.json]: Redact IP address.

File: test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+def test_withsegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(SEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS | SEG_DEPENDENT_LB_URIS
+
+
+def test_nosegid_lb(setup_neutronless_test, configure_icd, bigip):
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py]: Need to add a log scanner to look for ERROR messages upon test completion.  I believe this test should currently fail due to ERROR messages even if the correct URIs are created.

File: test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(NOSEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS
+
+
+@pytest.mark.skip(reason="This test will not pass until triggering the'\
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py]: This used to work, so please file a github issue and remove the 'skip' decorator so the test executes.

File: test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    logger.info(rdscache)
+    gtimeout = icontroldriver.conf.f5_network_segment_gross_timeout
+    poll_interval = icontroldriver.conf.f5_network_segment_polling_interval
+    # Set timers
+    start_time = time.time()
+    timeout = start_time + gtimeout
+    # Begin operations
+    while time.time() < (timeout - (2*poll_interval)):
+        time.sleep(poll_interval)
+        icontroldriver._common_service_handler(NOSEGID_CREATELB)
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py]: I don't think this really tests the feature.  You need to create the lb and listener with no segid, and sit in the while loop with NOSEGID_CREATELISTENER.

File: test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+    logger.setLevel(logging.ERROR)
+    logfilename = '/root/devenv/f5-openstack-agent/test/functional/'\
+                  'neutronless/l2adjacent/logtimeout.txt'
+    timeoutfh = logging.FileHandler(logfilename)
+    timeoutfh.setLevel(logging.ERROR)
+    logger.addHandler(timeoutfh)
+    # Begin operations
+    while time.time() < (timeout + (2*poll_interval)):
+        time.sleep(poll_interval)
+        icontroldriver._common_service_handler(NOSEGID_CREATELB)
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py]: Change NOSEGID_CREATELB to NOSEGID_CREATELISTENER.

File: test/functional/neutronless/l2adjacent/test_loadbalancer_create.py [hunk 9]
```diff
@@ -1,97 +0,0 @@
-# Copyright 2016 F5 Networks Inc.
```
Comment (mattgreene) [inline test/functional/neutronless/l2adjacent/test_loadbalancer_create.py]: Revert this change.  It's in liberty and we do not want to remove tests.

**Comment by mattgreene** [comment]
Done with review.  Comments inline...

File: test/functional/neutronless/l2adjacent/conf_symbols.json (hunk 2 above)
Comment (zancas) [inline test/functional/neutronless/l2adjacent/conf_symbols.json]: Reverted

File: test/functional/neutronless/l2adjacent/conf_symbols.json (hunk 3 above)
Comment (zancas) [inline test/functional/neutronless/l2adjacent/conf_symbols.json]: Reverted.

File: test/functional/neutronless/l2adjacent/oslo_confs.json (hunk 4 above)
Comment (zancas) [inline test/functional/neutronless/l2adjacent/oslo_confs.json]: Redacted.

File: test/functional/neutronless/l2adjacent/test_loadbalancer_create.py (hunk 9 above)
Comment (zancas) [inline test/functional/neutronless/l2adjacent/test_loadbalancer_create.py]: That module has been refactored into `neutronless/CD/test_loadbalancer.py`

**Comment by zancas** [comment]
@mattgreene review comments addressed.
